### ISSUE TITLE
In `emacs-lisp-mode`: `company-doc-buffer` (C-h while company is active): "user-error: No documentation available"

### ISSUE THREAD

**Issue by braham-snyder** [syl20bnr/spacemacs#8552]
#### Description 

On the develop branch, `company-doc-buffer` (`C-h` while the company popup is active) triggers `user-error: No documentation available` in `emacs-lisp-mode` (instead of showing documentation). Occurs for me ever since commit 8306626504817f3bb58eaceb8f4ca88ff3e9e5b5.

#### Reproduction guide 
```
mkdir ~/spacemacs
git clone https://github.com/syl20bnr/spacemacs.git ~/spacemacs/.emacs.d
cd spacemacs/.emacs.d
git checkout develop
HOME=~/spacemacs emacs # or env HOME=$HOME/spacemacs emacs
```
- Uncomment `auto-completion` layer from the dotfile, then `SPC f e R`, then restart (should the restart actually be necessary? completion tooltip wasn't popping up for me without it, but I think that's unrelated to this issue)
- Open a scratch buffer
- Enable `emacs-lisp-mode`
- Insert a completable substring, e.g., `(mes|)`
- Press `C-h` to call `company-doc-buffer`

*Observed behaviour:* 
- `user-error: No documentation available`

*Expected behaviour:* 
- A new window showing the documentation for the selected company candidate (the first candidate, by default) pops up

#### System Info 
- OS: darwin
- Emacs: 25.1.1
- Spacemacs: 0.200.7
- Spacemacs branch: develop (rev. c9d332fb)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm auto-completion emacs-lisp)
```
- System configuration features: JPEG RSVG NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES

**Comment by bmag** [comment]
Tested and I get the same error. If I disable `purpose-mode`, then I am able to view documentation. Worth noting this happens to me in Elisp buffers but not in Python buffers. I'll have to investigate this, but I don't know when I will get the time... (with luck it will happen this Friday)

**Comment by cmatzenbach** [comment]
For what it's worth, I'm a vanilla emacs user googling for this exact same error right now.  I am having this problem in elisp, javascript (tern) and typescript (tide) modes.

**Comment by Miciah** [comment]
This is a duplicate of issue #9593 and is resolved by #9594 / https://github.com/syl20bnr/spacemacs/commit/ca28bde0497d5c68ceeb4d943630d91caa79b587.

The cause of the issue was that `elisp--company-doc-buffer` would invoke `describe-function`, which would open a `*Help*` buffer, which popwin would immediately close, which would cause `elisp--company-doc-buffer` to return `nil`, which would cause company to conclude that no documentation could be found.
